### PR TITLE
[time]: Document all functions

### PR DESCRIPTION
**Pull request by flysand7**
This is going to be the first PR in the series where I go over each package and make a consistent way in which functions are documented, so I might as well describe what exactly I'm doing here.

In this PR I wrote documentation for most public procedures, constants and structs in the `core:time` package, which may be incorrect / worded poorly / inconsistent, so review is appreciated.

# Description of the format used

The style is currently trying to be consistent with `core:strings`, so what I'm describing for the most part is that style + some flavors.

1. All documentation of top-level constructs (procedures, types, constants) is done with block comments
2. The documentation text is not indented (tools that treat documentation as markdown will confuse this indentation for a code block, making none of the formatting appear correctly)
3. The documentation is written using standard markdown

The body of the documentation text is either
1. A single line of summary
2. (optionally) Full description that may repeat or expand upon summary (will appear duplicated)
3. (optionally) Documentation of input and output parameters of the procedure
4. (optionally) Examples and notes

The documentation of input and output parameters would look pretty much always like this:

```
**Inputs**:
- `param_name`: Param Description
- `param_name`: Param Description

**Returns**:
- Description of unnamed return, if named its just like the above, if multiple returns use a numbered list
```

The documentation of notes has the following format:

```
**Note**: A single note
**Note(windows)**: A platform-specific note

**Notes**:
- List of notes
- Note2
- Note3
```

Unindented documentation doesn't look pretty but we may have to live with that.

### CODE REVIEW

File: core/time/rfc3339.odin [hunk 1]
```diff
@@ -4,10 +4,33 @@ package time
...
+Parse an RFC 3339 string into time with a UTC offset applied to it.
+
+This procedure parses the specified RFC 3339 strings of roughly the following
+format:
+
+```text
+YYYY-MM-DD[Tt]HH:mm:ss[.nn][Zz][+-]HH:mm
+```
+
+And returns the time that was represented by the RFC 3309 string, with the UTC
```
Comment (jonasgf) [inline core/time/rfc3339.odin:17]: RFC 3309 -> RFC 3339. This typo is repeated a few places.

File: core/time/rfc3339.odin [hunk 2]
```diff
@@ -38,9 +84,31 @@ rfc3339_to_time_and_offset :: proc(rfc_datetime: string, is_leap: ^bool = nil) -
 	}
 }
 
-// Parses an RFC 3339 string and returns Time and a UTC offset in minutes.
-// e.g. 1985-04-12T23:20:50.52Z
-// Performs no validation on whether components are valid, e.g. it'll return hour = 25 if that's what it's given
+/*
+Parse an RFC 3339 string into a datetimt and a UTC offset in minutes.
```
Comment (jonasgf) [inline core/time/rfc3339.odin:88]: datetimt -> datetime

File: core/time/rfc3339.odin [hunk 3]
```diff
@@ -38,9 +84,31 @@ rfc3339_to_time_and_offset :: proc(rfc_datetime: string, is_leap: ^bool = nil) -
...
+
+This procedure parses the specified RFC 3339 strings of roughly the following
+format:
+
+```text
+YYYY-MM-DD[Tt]HH:mm:ss[.nn][Zz][+-]HH:mm
+```
+
+And returns the datetime, in UTC and the utc offset, in minutes, that were
```
Comment (jonasgf) [inline core/time/rfc3339.odin:97]: utc -> UTC

File: core/time/iso8601.odin [hunk 4]
```diff
@@ -3,23 +3,62 @@ package time
...
+And returns time, in UTC represented by that string. In case the timezone offset
+is specified in the string, that timezone is applied to time.
+
+**Inputs**:
+- `iso_datetime`: The string to be parsed
+- `is_leap`: Optional output parameter, specifying if the moment was a leap second
+
+**Returns**:
+- `res`: The time represented by `iso_datetime`, with UTC offset applied
+- `consumed`: The number of character that were consumed from the string
```
Comment (jonasgf) [inline core/time/iso8601.odin:24]: character -> characters

File: core/time/iso8601.odin [hunk 5]
```diff
@@ -3,23 +3,62 @@ package time
...
+minutes.
+
+**Inputs**:
+- `iso_datetime`: The string to be parsed
+- `is_leap`: Optional output parameter, specifying if the moment was a leap second
+
+**Returns**:
+- `res`: The time in UTC
+- `utc_offset`: The UTC offset of the time, in minutes
+- `consumed`: The number of character that were consumed from the string
```
Comment (jonasgf) [inline core/time/iso8601.odin:56]: character -> characters

**Comment by flysand7** [comment]
@jonasgf thanks for review, the fixes have been made. Force-pushed the branch to not bloat commit history with little edits

**Comment by flysand7** [comment]
Spurious CI error

**Pull request merged**